python/pyarrow/_s3fs.pyx

```diff
@@ -168,6 +180,31 @@ cdef class S3FileSystem(FileSystem):
         if background_writes is not None:
             options.background_writes = background_writes
 
+        if proxy_options is not None:
+            if isinstance(proxy_options, dict):
+                proxy_scheme = proxy_options.get("scheme", None)
```

Comment:
If this is required than we should simply use `proxy_options["scheme"]`, no?
Same below for "host" and "port".

Reply:
Fair point. Will make the change.

Reply:
Done.